From 12ec1a9bea00bd585d86600b3f93cd617b065cb0 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 19 Sep 2024 13:45:56 +0200
Subject: [PATCH 1241/1695] clk: stm32mp25: set FMC clock as critical

To simplify the clock management of FMC, it has been decided to only allow the TDCID to handle the FMC clock.
Until, TF-M does not have a SCMI agent, set the FMC clock as critical

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6e461989d06ef86571c0594dc80b433e14620890 clk: stm32mp25: set FMC clock as critical]
Change-Id: I5427f76c99bf03c21d07b34a174b49330b491b47
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/405650
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 5aaee22cd..98463f13a 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -4348,6 +4348,7 @@ static bool clk_stm32_clock_is_critical(struct clk *clk)
 		&ck_icn_p_ddrphyc,
 		&ck_icn_s_sysram,
 		&ck_icn_s_bkpsram,
+		&ck_ker_fmc,
 		&ck_ker_ospi1,
 		&ck_ker_ospi2,
 		&ck_icn_s_vderam,
-- 
2.47.2

